src/main/scala/firrtl_interpreter/CircuitState.scala

```diff
@@ -116,19 +123,25 @@ case class CircuitState(
     * cycle all memories
     */
   def cycle(): Unit = {
+    vcdLoggerOption.foreach { vcd =>
+      vcd.incrementTime(10)
+      vcd.raiseClock
+    }
     registers.keys.foreach { key =>
       registers(key) = nextRegisters(key)
     }
 //    nextRegisters.clear()
 //    ephemera.clear()
     cycleMemories()
-    vcdLoggerOption.foreach { vcd =>
-      vcd.incrementTime()
-    }
     nameToConcreteValue = mutable.HashMap((inputPorts ++ outputPorts ++ registers).toSeq:_*)
     isStale = true
 
     stateCounter += 1
+
+    vcdLoggerOption.foreach { vcd =>
+      vcd.incrementTime(10)
```

Comment:
I guess [https://github.com/ucb-bar/firrtl-interpreter/blob/38cf05bee320d9802f8d662cc02315a3d7047607/src/main/scala/firrtl_interpreter/vcd/VCD.scala#L541](url) needs to be fixed to `timeStamp += increment`